Model/RatepayInvoicePaymentMethod.php

```diff
@@ -461,4 +473,133 @@ private function _areAddressesAllowed($sBillingCountryId, $sShippingCountryId)
             $sBillingCountryId === $sShippingCountryId
         );
     }
+
+    /**
+     *
+     * @inheritdoc
+     *
+     * @param string                           $sAction
+     * @param \Wirecard\Oxid\Model\Transaction $oParentTransaction
+     * @param array|null                       $aOrderItems
+     *
+     * @return Transaction
+     *
+     * @since 1.2.0
+     */
+    public function getPostProcessingTransaction($sAction, $oParentTransaction, $aOrderItems = null)
+    {
+        $oTransaction = new RatepayInvoiceTransaction();
+        $aBasketResult = $this->_createPostProcessingBasket($oParentTransaction, $aOrderItems);
+
+        /**
+         * @var $oBasket Basket
+         */
+        $oBasket = $aBasketResult[0];
+        $oTransaction->setBasket($oBasket);
+        $oTransaction->setAmount(new Amount(
+            $aBasketResult[1],
+            $oBasket->getTotalAmount()->getCurrency()
+        ));
+        return $oTransaction;
+    }
+
+    /**
+     * @param \Wirecard\Oxid\Model\Transaction $oParentTransaction
+     * @param array|null                       $aOrderItems
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    private function _createPostProcessingBasket($oParentTransaction, $aOrderItems)
+    {
+        $oBasket = new Basket();
+
+        $oXmlBasket = simplexml_load_string($oParentTransaction->getResponseXML());
+        $oBasket->parseFromXml($oXmlBasket);
+
+        $aItemsToAdd = self::_getItemsToAddToBasket($oBasket, $oXmlBasket, $aOrderItems);
+
+        $oPostProcBasket = new Basket();
+        $oPostProcBasket->setVersion(RatepayInvoiceTransaction::class);
+
+        $iRoundPrecision = Helper::getCurrencyRoundPrecision($oBasket->getTotalAmount()->getCurrency());
+
+        $fAmount = 0;
+        foreach ($aItemsToAdd as $oBasketItem) {
+            /**
+             * @var $oBasketItem Item
+             */
+            $oPostProcBasket->add($oBasketItem);
+            $fAmount = round(
+                bcadd(
+                    $fAmount,
+                    $oBasketItem->getPrice()->getValue() * $oBasketItem->getQuantity(),
+                    Helper::BCSUB_SCALE
+                ),
+                $iRoundPrecision
+            );
+        }
+
+        return [$oPostProcBasket, $fAmount];
+    }
+
+    /**
+     * @param Basket            $oBasket
+     * @param \SimpleXMLElement $oXmlBasket
+     * @param array             $aOrderItems
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    private static function _getItemsToAddToBasket($oBasket, $oXmlBasket, $aOrderItems)
+    {
+        $aItemsToAdd = [];
+
+        foreach ($aOrderItems as $sArticleNumber => $iQuantity) {
+            if ($iQuantity < 1) {
```

Comment:
Why are even items with less quantity here, if you filter them before you save one loop

Reply:
that is the point where I'm filtering.